cpp/src/arrow/filesystem/path_util.h

```diff
@@ -55,6 +60,11 @@ Status ValidateAbstractPathParts(const std::vector<std::string>& parts);
 ARROW_EXPORT
 std::string ConcatAbstractPath(const std::string& base, const std::string& stem);
 
+// Append a non-empty stem to an abstract path with a filename prefix.
+ARROW_EXPORT
+std::string ConcatAbstractPath(const std::string& base, const std::string& prefix,
```

Comment:
I think this method is a little too single purpose.  This file is in the `filesystem` module and is meant to contain general purpose utilities that are common for anyone working with files.  There is no concept of "partitions" or "prefixes" in the domain of this module.  Perhaps you could instead create a more general version:

```
std::string ConcatAbstractPaths(const std::vector<std::string>& parts);
```

Reply:
Made a generic ConcatAbstractPaths()